File: pdlearning/src/web-apps/src/libraries/domain-api/src/learner/models/user-review.model.ts
import { PublicUserInfo, UserInfoModel } from '../../share/models/user-info.model';

import { UserReviewItemType } from '../../content/models/user-review-item-type.enum';

export interface IUserReviewModel {
  id?: string;
  userId: string;
  itemId: string;
  classRunId?: string;
  itemType: UserReviewItemType;
  parentCommentId?: string | undefined;
  version?: string;
  itemName?: string | undefined;
  userFullName?: string;
  commentTitle?: string | undefined;
  commentContent?: string | undefined;
  rate?: number;
  createdDate?: Date;
  createdBy?: string;
  changedDate?: Date | undefined;
  changedBy?: string | undefined;
}

export class UserReviewModel implements IUserReviewModel {
  public id: string;
  public userId: string;
  public itemId: string;
  public classRunId?: string;
  public itemType: UserReviewItemType;
  public parentCommentId?: string | undefined;
  public version?: string | undefined;
  public itemName?: string | undefined;
  public userFullName?: string;
  public commentTitle?: string | undefined;
  public commentContent?: string | undefined;
  public rate?: number;
  public createdDate: Date;
  public createdBy?: string;
  public changedDate?: Date | undefined;
  public changedBy?: string | undefined;
  public user: PublicUserInfo;

  constructor(data?: IUserReviewModel) {
    if (data == null || data === null) {
      return;
    }
    this.id = data.id;
    this.userId = data.userId;
    this.itemId = data.itemId;
    this.classRunId = data.classRunId;
    this.itemType = data.itemType;
    this.parentCommentId = data.parentCommentId;
    this.version = data.version;
    this.itemName = data.itemName;
    this.userFullName = data.userFullName;
    this.commentTitle = data.commentTitle;
    this.commentContent = data.commentContent;
    this.rate = data.rate;
    this.createdDate = new Date(data.createdDate);
    this.createdBy = data.createdBy;
    this.changedDate = data.changedDate !== undefined ? new Date(data.changedDate) : undefined;
    this.changedBy = data.changedBy;
  }

  public get isMine(): boolean {
    return this.userId === UserInfoModel.getMyUserInfo().extId;
  }
}

File: pdlearning/src/microservices/learner-microservice/Api/Application/Models/UserReviewModel.cs
using System;
using Microservice.Learner.Domain.Entities;
using Microservice.Learner.Domain.ValueObject;

namespace Microservice.Learner.Application.Models
{
    public class UserReviewModel
    {
        public UserReviewModel()
        {
        }

        public UserReviewModel(UserReview entity)
        {
            Id = entity.Id;
            ParentCommentId = entity.ParentCommentId;
            UserId = entity.UserId;
            ItemId = entity.ItemId;
            ItemType = entity.ItemType;
            Version = entity.Version;
            ItemName = entity.ItemName;
            UserFullName = entity.UserFullName;
            CommentTitle = entity.CommentTitle;
            CommentContent = entity.CommentContent;
            Rate = entity.Rate;
            IsDeleted = entity.IsDeleted;
            CreatedBy = entity.CreatedBy;
            CreatedDate = entity.CreatedDate;
            ChangedBy = entity.ChangedBy;
            ChangedDate = entity.ChangedDate;
            ClassRunId = entity.ClassRunId;
        }

        public Guid Id { get; set; }

        public Guid? ParentCommentId { get; set; }

        public Guid UserId { get; set; }

        public Guid ItemId { get; set; }

        public ItemType ItemType { get; set; }

        public string Version { get; set; }

        public string ItemName { get; set; }

        public string UserFullName { get; set; }

        public string CommentTitle { get; set; }

        public string CommentContent { get; set; }

        public double? Rate { get; set; }

        public bool IsDeleted { get; set; }

        public Guid CreatedBy { get; set; }

        public Guid? ChangedBy { get; set; }

        public DateTime CreatedDate { get; set; }

        public DateTime? ChangedDate { get; set; }

        public Guid? ClassRunId { get; set; }
    }
}
